doc/lua_api.txt

```diff
@@ -2201,13 +2201,17 @@ appearing when you don't expect them to. See `no_prepend[]`
 
 #### `textarea[<X>,<Y>;<W>,<H>;<name>;<label>;<default>]`
 * Same as fields above, but with multi-line input
-* if the text overflows a vertical scrollbar is added
-* if the name is empty the textarea is readonly, the label is not displayed.
+* If the text overflows, a vertical scrollbar is added.
+* If the name is empty, the textarea is readonly and
+  the background is not shown, which corresponds to a multi-line label.
 
 #### `label[<X>,<Y>;<label>]`
 * `x` and `y` work as per field
-* `label` is the text on the label
-* Position and size units are inventory slots
+* The label formspec element displays the text set in `label`
+  at the specified position.
+* For text with multiple lines, refer to textarea.
```

Comment:
Don't talk about textarea in label section. Talk about textarea in textarea section.

Reply:
Do you think that modders can easily find out about how to put multi-line text?

Reply:
You need to make it explicit that label is for single line instead. It's even fine to state it's not for multi-line use. Then if we make sure the textarea section mentions usage includes multi-line, the problem is gone. This way each section is explaining its own use and limits, and not explaining other things' uses and limits.